src/cluster_legacy.c

```diff
@@ -6552,3 +6552,30 @@ void clusterReplicateOpenSlots(void) {
 
     zfree(argv);
 }
+
+del_stat delUnOwnedKeys(void) {
+    del_stat result = {0, 0};
+    if (!server.cluster_enabled) {
+        return result;
+    }
+
+    clusterNode *master;
```

Comment:
If we're not a primary (master) we should not be taking any actions here. We should let the primary do the deletions and the replica should follow it. The replica needs to be a carbon copy of the primary, and there are cases where the replica can think it's primary doesn't own a slot it actually does, and won't clean up the data.

Reply:
Will replication sync remove orphaned keys?

Reply:
Looks like replica will flush old data before loading rdb by calling emptyData()

Reply:
Think it second time, I think do it for both replica and primary is a safer way. Consider replica doesn't remove the keys, then before sync with primary, it is promoted to be new primary. Then these data will not be removed.